sjsonnet/src/sjsonnet/Parser.scala

```diff
@@ -160,18 +141,37 @@ class Parser(
     }
   ).map(_.mkString)
 
-  def tripleBarStringHead[$: P]: P[(Seq[String], String, String)] = P(
-    (CharsWhileIn(" \t", 0) ~~ "\n".!).repX ~~
-    CharsWhileIn(" \t", 1).! ~~
-    CharsWhile(_ != '\n').!
+  def maybeChompedTripleBarString[$: P]: P[Seq[String]] = tripleBarString.map {
+    case (true, lines) =>
+      Seq(lines.mkString.stripLineEnd)
+    case (false, lines) =>
+      lines
+  }
+
+  def tripleBarString[$: P]: P[(Boolean, Seq[String])] = P(
+    ("||-" | "||").!.map(_.last == '-')./ ~~ CharsWhileIn(" \t", 0) ~~
+    // Detect the new line separator
+    ("\r\n" | "\n").!./.flatMapX { sep =>
```

Comment:
if it fail, we should give an error message, the `text block must start with at least one whitespace` here